sdk/mobile-center-distribute/src/main/java/com/microsoft/azure/mobile/distribute/Distribute.java

```diff
@@ -544,6 +555,21 @@ private synchronized void resumeDistributeWorkflow() {
                 return;
             }
 
+            // If failed to enable in-app updates on the same app build before, don't go any further.
+            // Only if the app build is different (different package hash), try enabling in-app updates again.
+            String releaseHash = DistributeUtils.computeReleaseHash(this.mPackageInfo);
+            String updateSetupFailedPackageHash = PreferencesStorage.getString(PREFERENCE_KEY_UPDATE_SETUP_FAILED_PACKAGE_HASH_KEY);
+            if (updateSetupFailedPackageHash != null) {
+                if (releaseHash.equals(updateSetupFailedPackageHash)) {
+                    MobileCenterLog.info(LOG_TAG, "Skipping in-app updates setup, because it already failed on this release before.");
+                    return;
+                } else {
+                    MobileCenterLog.info(LOG_TAG, "Re-attempting in-app updates setup and cleaning up failure info from storage.");
+                    PreferencesStorage.remove(PREFERENCE_KEY_UPDATE_SETUP_FAILED_PACKAGE_HASH_KEY);
```

Comment:
When SDK is disabled you need to remove those new keys as well (in the method where everything is cleared).

Reply:
This comment is still active

Reply:
To be more accurate in `applyEnabledState(false)` : remove the new keys as well in the else.